### PR TITLE
add functions for SQS and S3

### PR DESCRIPTION
**Pull request by jingpengwu**
- functions to use SQS directly
- list S3 objects
- download S3 file

### CODE REVIEW

**Comment by mdpradeep** [comment]
Thanks @ jingpengwu for this. Is it possible to add a few test cases for these new APIs ? It would be really helpful.

**Comment by mdpradeep** [comment]
Also, some of the existing tests are failing https://github.com/JuliaCloud/AWS.jl/blob/master/test/tests3.jl & https://github.com/JuliaCloud/AWS.jl/blob/master/test/testsqs.jl. Could you check them too ?

File: src/AWS.jl [hunk 1]
```diff
@@ -40,6 +40,19 @@ if haskey(ENV, "AWS_ID") && haskey(ENV, "AWS_SECKEY")
...
+        line = replace(line, "\n", "")
+        segs = split(line, "=")
+        if ismatch(r"^aws_secret_access_key", segs[1])
+            AWS_SECKEY = segs[2]
+        elseif ismatch(r"^aws_access_key_id", segs[1])
+            AWS_ID = segs[2]
+        elseif ismatch(r"^region", segs[1])
+            AWS_REGION = segs[2]
+        end
+    end
```
Comment (samoconnor) [inline src/AWS.jl]: Need to handle multiple profiles in config file.
e.g. https://github.com/samoconnor/AWSCore.jl/blob/master/src/AWSCredentials.jl#L173

http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-config-files

```ini
[default]
aws_access_key_id = XXX
aws_secret_access_key = XXX
[other]
aws_access_key_id = XXX
aws_secret_access_key = XXX
```

File: src/AWS.jl [hunk 2]
```diff
@@ -40,6 +40,19 @@ if haskey(ENV, "AWS_ID") && haskey(ENV, "AWS_SECKEY")
 elseif haskey(ENV, "AWS_ACCESS_KEY_ID") && haskey(ENV, "AWS_SECRET_ACCESS_KEY")
     AWS_ID = ENV["AWS_ACCESS_KEY_ID"]
     AWS_SECKEY = ENV["AWS_SECRET_ACCESS_KEY"]
+elseif isfile(joinpath(homedir(), ".aws/config"))
```
Comment (samoconnor) [inline src/AWS.jl]: `.aws/credentials` ? (not `config`)

Comment (jingpengwu) [inline src/AWS.jl]: I have both. I used to use starcluster.

File: src/AWS.jl (hunk 1 above)
Comment (jingpengwu) [inline src/AWS.jl]: Since most of people only have one account in his/her own home folder, this might be a special case.

**Comment by jingpengwu** [comment]
just updated the code and solved some conflicts. tested the s3 and sqs.

**Comment by mdpradeep** [comment]
@jingpengwu, sorry for the delay, but can you please rebase this against master ?

**Comment by jingpengwu** [comment]
just rebased, please take a look.

**Comment by jingpengwu** [comment]
@mdpradeep Sorry, all of the S3 functions do not work. This might be caused by this PR or another recent PR(https://github.com/JuliaCloud/AWS.jl/pull/79), which seems unlikely. 

 The `tests3.jl` have S3Errors. I can not find any problem of this PR though.

**Pull request merged**